### PR TITLE
rollout apis

### PR DESCRIPTION
**Pull request by zmberg**

### CODE REVIEW

File: api/v1alpha1/rollout_types.go [hunk 1]
```diff
@@ -46,40 +66,54 @@ type ObjectRef struct {
...
 
 // CanaryStep defines a step of a canary workload.
 type CanaryStep struct {
 	// SetWeight sets what percentage of the canary pods should receive
-	SetWeight *int32 `json:"setWeight,omitempty"`
+	Weight int32 `json:"weight,omitempty"`
+	// Replicas is the number of expected canary pods in this batch
+	// it can be an absolute number (ex: 5) or a percentage of total pods.
+	// it is mutually exclusive with the PodList field
+	CanaryReplicas *intstr.IntOrString `json:"canaryReplicas,omitempty"`
```
Comment (furykerry) [inline api/v1alpha1/rollout_types.go:112]: rename CanaryReplicas to just Replicas ?

File: api/v1alpha1/rollout_types.go [hunk 2]
```diff
@@ -46,40 +66,54 @@ type ObjectRef struct {
...
 }
 
 // CanaryStep defines a step of a canary workload.
 type CanaryStep struct {
 	// SetWeight sets what percentage of the canary pods should receive
-	SetWeight *int32 `json:"setWeight,omitempty"`
+	Weight int32 `json:"weight,omitempty"`
+	// Replicas is the number of expected canary pods in this batch
+	// it can be an absolute number (ex: 5) or a percentage of total pods.
+	// it is mutually exclusive with the PodList field
```
Comment (furykerry) [inline api/v1alpha1/rollout_types.go:111]: what is the. PodList field?

File: api/v1alpha1/rollout_types.go [hunk 3]
```diff
@@ -46,40 +66,54 @@ type ObjectRef struct {
...
 type RolloutStrategy struct {
+	// Paused indicates that the Rollout is paused.
+	// Default value is false
+	Paused bool `json:"paused,omitempty"`
+	// canaryPlan, BlueGreenPlan
+	// Default value is canaryPlan
+	Type RolloutStrategyType `json:"type"`
 	// +optional
-	// BlueGreen *BlueGreenStrategy `json:"blueGreen,omitempty" protobuf:"bytes,1,opt,name=blueGreen"`
+	CanaryPlan *CanaryStrategy `json:"canaryPlan,omitempty"`
```
Comment (furykerry) [inline api/v1alpha1/rollout_types.go:83]: `json:"canaryPlan,omitempty"` just `json:"canary,omitempty"`

**Review by furykerry — COMMENTED** [review]
/lgtm

**Review by FillZpp — APPROVED** [review]
/approve

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**